### PR TITLE
Move device enumeration to device specific drivers and load device drivers without explicit import

### LINKED ISSUE #55: WebUSB support (e.g. Trezor model T)
The Trezor model T uses [webusb](https://wicg.github.io/webusb/) to communicate (see https://github.com/achow101/HWI/issues/38#issuecomment-427032492). I believe Ledger Nano S [supports U2F](https://github.com/LedgerHQ/ledger-wallet-chrome/issues/88#issuecomment-363222976).

I'm not entirely clear on the differences between webusb and U2F.

I'm guessing trying and studying https://github.com/trezor/python-trezor would be a good place to start.

### PR DESCRIPTION
**Pull request by achow101**
This PR moves device enumeration from the `enumerate()` function to `enumerate()` functions within each device driver. These device specific `enumerate()` functions allow us to enumerate those devices with whatever device specific enumeration required, instead of only enumerating with hid. This allows us to interact with devices connected through other methods such as udp (trezor emulator), webusb (trezor t), unix pipes (coldcard emulator), etc. but without requiring us to add that enumeration to the actual `enumerate()` function. Because there can be different types of device interfaces, the creation of the device is now handled by the device client class. The device path will be passed in and the client class uses that to know what type of device interface to create.

Furthermore, device drivers have been moved to `hwilib/devices/` and both `getclient()` and `enumerate()` will call their respective functions and constructors without explicitly importing the device specific drivers. This allows us to easily add a new device driver by simply adding its implementation to the `hwilib/devices/` folder. In order to facilitate this automatic discovery and import of device drivers, the client class names and arguments have been normalized. The class name must be the device name (as will be specified in the `--device-type` parameter), begin with a capital letter, have the rest of the device name in lower case, and followed by `Client`. Classes must also take two positional arguments `path` and `password`, in that order.

This change should enable the use of the Trezor T (so fixes #55) as well as device simulators and emulators so that automated tests can be done.

### CODE REVIEW

File: hwilib/commands.py [hunk 1]
```diff
@@ -32,30 +32,24 @@ def __init__(self,*args,**kwargs):
 
 # Get the client for the device
 def get_client(device_type, device_path, password=None):
-    # Open the device
```
Comment (instagibbs) [inline hwilib/commands.py]: don't think these lines should be deleted yet in this commit?

Comment (achow101) [reply hwilib/commands.py]: Moved to "Unify device client constructor to path and password "

File: hwilib/devices/ledger.py [hunk 2]
```diff
@@ -1,26 +1,33 @@
...
 import binascii
 import logging
 
+LEDGER_VENDOR_ID = 0x2c97
+LEDGER_DEVICE_ID = 0x0001
+
 # This class extends the HardwareWalletClient for Ledger Nano S specific things
 class LedgerClient(HardwareWalletClient):
 
     # device is an HID device that has already been opened.
```
Comment (instagibbs) [inline hwilib/devices/ledger.py]: you can delete these stale lines, especially now the initialization is done locally.

File: hwilib/devices/coldcard.py [hunk 3]
```diff
@@ -106,3 +115,34 @@ def wipe_device(self):
...
+            master_xpub = client.get_pubkey_at_path('m/0h')['xpub']
+            d_data['fingerprint'] = get_xpub_fingerprint_hex(master_xpub)
+            client.close()
+        except Exception as e:
+            d_data['error'] = "Could not open client or get fingerprint information: " + str(e)
+
+        results.append(d_data)
+    # Check if the simulator is there
+    try:
+        client = ColdcardClient(CC_SIMULATOR_SOCK)
```
Comment (instagibbs) [inline hwilib/devices/coldcard.py]: Can a tutorial on setting up the simulator be linked?

Comment (achow101) [reply hwilib/devices/coldcard.py]: Some documentation for it is here: https://github.com/Coldcard/firmware/blob/master/unix/README.md

Although I haven't quite figured it out for myself either.

I added a comment with a link to the docs.

File: hwilib/commands.py [hunk 4]
```diff
@@ -22,82 +24,35 @@
...
-        except Exception as e:
-            d_data['error'] = "Could not open client or get fingerprint information: " + str(e)
-            pass
+    # Gets the module names of all the files in devices/
+    files = glob.glob(dirname(__file__)+"/devices/*.py")
+    modules = [ basename(f)[:-3] for f in files if isfile(f) and not f.endswith('__init__.py')]
 
-        result.append(d_data)
+    for module in modules:
+        imported_dev = importlib.import_module('.devices.' + module, __package__)
```
[the remark this answers is not in the document]
Comment (achow101) [reply hwilib/commands.py]: Done. the errors should be suppressed now.

File: hwilib/devices/ledger.py (hunk 2 above)
Comment (achow101) [reply hwilib/devices/ledger.py]: Removed

**Comment by instagibbs** [comment]
Ok Ledger stuff at least works now.

File: hwilib/commands.py [hunk 5]
```diff
@@ -53,41 +53,12 @@ def get_client(device_type, device_path, password=None):
 # Get a list of all available hardware wallets
 def enumerate(args):
```
Comment (Sjors) [inline hwilib/commands.py]: It would be nice if `enumerate` didn't ignore `--device-type=` as it's a bit slow.

Comment (instagibbs) [reply hwilib/commands.py]: I like the suggestion but obviously should wait until this is merged

File: hwilib/devices/trezor.py [hunk 6]
```diff
@@ -195,3 +193,22 @@ def wipe_device(self):
     # Close the device
     def close(self):
         self.client.close()
+
```
[the remark this answers is not in the document]
Comment (achow101) [reply hwilib/devices/trezor.py]: Done

**Pull request merged**